src/util-inl.h

```diff
@@ -404,7 +404,7 @@ inline char* UncheckedCalloc(size_t n) { return UncheckedCalloc<char>(n); }
 void ThrowErrStringTooLong(v8::Isolate* isolate);
 
 v8::MaybeLocal<v8::Value> ToV8Value(v8::Local<v8::Context> context,
-                                    const std::string& str,
+                                    std::string_view str,
```

Comment:
This would mean that we have to duplicate a large amount of headers in our code base. I would not want to do that without automated linting that keeps things in sync.

Reply:
But then is there a point of having https://github.com/nodejs/node/blob/master/doc/contributing/cpp-style-guide.md? I don't think `make lint-cpp` enforces even half of the rules we write in the doc.

I haven't ever tried but I think https://github.com/include-what-you-use/include-what-you-use might be able to automate this. FYI there has been a lot of IWYU improvements in chromium in the last 8 months and it's had a noticeable impact on the total CPU time for a build - https://commondatastorage.googleapis.com/chromium-browser-clang/build-time.html.

Reply:
> But then is there a point of having https://github.com/nodejs/node/blob/master/doc/contributing/cpp-style-guide.md? I don't think `make lint-cpp` enforces even half of the rules we write in the doc.

Yeah, that second part is the reason why we have the style guide. We don’t have a linter that’s good enough.

This is different from a general linting rule, though; this is something that would require a lot of unnecessary manual labor if the includes are not kept in sync automatically.

> I haven't ever tried but I think https://github.com/include-what-you-use/include-what-you-use might be able to automate this. 

Sure – feel free to give it a try on the Node.js codebase!

> FYI there has been a lot of IWYU improvements in chromium in the last 8 months and it's had a noticeable impact on the total CPU time for a build - https://commondatastorage.googleapis.com/chromium-browser-clang/build-time.html.

That makes me happy for the Chromium team, but seems to have no impact on this discussion? It’s not like I’m disagreeing with IWYU as a principle.

Reply:
> This is different from a general linting rule, though; this is something that would require a lot of unnecessary manual labor if the includes are not kept in sync automatically.

Hmm, that's true.

> Sure – feel free to give it a try on the Node.js codebase!

Will do!

> That makes me happy for the Chromium team, but seems to have no impact on this discussion? It’s not like I’m disagreeing with IWYU as a principle.

Since I suggested using it in this pr, I thought it might be essential to at least provide some info about its impact.